src/github.com/couchbase/sync_gateway/db/design_doc.go

```diff
@@ -108,7 +124,7 @@ func (db *Database) QueryDesignDoc(ddocName string, viewName string, options map
 	// Query has slightly different access control than checkDDocAccess():
 	// * Admins can query any design doc including the internal ones
 	// * Regular users can query non-internal design docs
-	if db.user != nil && isInternalDDoc(ddocName) {
+	if db.user != nil && (isInternalDDoc(ddocName) || !db.GetUserViewsEnabled()) {
```

Comment:
this is a bit hard to mentally parse, what about separate guards?

```
if !db.GetUserViewsEnabled() {
return nil, base.HTTPErrorf(http.StatusForbidden, "forbidden")
}

if db.user != nil && isInternalDDoc(ddocName) { 
return nil, base.HTTPErrorf(http.StatusForbidden, "forbidden")
}
```

I realize it's a bit more code and a line is duplicated, but I think it's easier to understand.